gson/src/test/java/com/google/gson/JsonObjectTest.java

```diff
@@ -163,6 +163,7 @@ public void testEqualsOnEmptyObject() {
   }
 
   @Test
+  @SuppressWarnings("TruthSelfEquals")
```

Comment:
This should not be suppressed, see [its description](https://errorprone.info/bugpattern/TruthSelfEquals).